### DISCUSSION TITLE
Semantic Kernel use of Vector Store vs LLM

### DISCUSSION THREAD

**Discussion by dsteketee76** [microsoft/semantic-kernel#3164] [General]
I'm trying to better understand how the SK incorporates content from a vector store (either volatile or persistent) vs from the core LLM. I've observed responses which are intended to interrogate content from embeddings return content from the core LLM as well as actively deny the presence of content from the embeddings. So, I'm curious what drives this behavior and if there is any way to address it.

**Comment by CharlieDigital** [comment]
> ...as well as actively deny the presence of content from the embeddings.

I had this issue, but found that when using the following in SK plugins invoked from my native plugins:

```
{{recall $heading}}
```

That I needed to specify the correct parameters:

```csharp
variables = new ContextVariables {
  [TextMemoryPlugin.CollectionParam] = topic.AsSha256Hex(),
  [TextMemoryPlugin.LimitParam] = "15", // Tune this up or down
  [TextMemoryPlugin.RelevanceParam] = "0.60",
  [nameof(topic)] = topic,
  [nameof(heading)] = heading,
  [nameof(previous)] = previous
};

var enrichListItem = _kernel.Functions.GetFunction(
  "ListGeneratorPlugin",
  "EnrichListItem"
);

var enriched = await _kernel.RunAsync(variables, enrichListItem);
```

Basically, without explicitly specifying the `CollectionParam`, it doesn't match results
